lang_GENERIC/analyze/Constant_propagation.ml

```diff
@@ -357,5 +358,5 @@ let propagate_dataflow ast =
   v (Pr ast)
 
 let propagate a b = Common.profile_code "Constant_propagation.xxx" (fun () ->
-  propagate2 a b
-)
+  propagate_basic a b;
```

Comment:
What about we get rid of the generate propagate and have propagate_basic and propagate_dataflow
and we call both in semgrep at some point. That way we can merge already this in pfff and semgrep,
and only at some point later we just add a call in semgrep-core/parsing/Parse_code.parse_and_resolve_name_...
to Constant_propagation.propagate_dataflow in addition to propagate_basic.

Reply:
Great idea, done.